kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/NetworkAdminServiceImpl.java

```diff
@@ -945,13 +946,48 @@ public void enableInterface(String interfaceName, boolean dhcp) throws KuraExcep
         try {
             NetInterfaceType type = LinuxNetworkUtil.getType(interfaceName);
 
-            if (!LinuxNetworkUtil.hasAddress(interfaceName)
-                    || type == NetInterfaceType.WIFI && !LinuxNetworkUtil.isLinkUp(interfaceName)) {
+            NetInterfaceStatus status = NetInterfaceStatus.netIPv4StatusUnknown;
+            WifiMode wifiMode = WifiMode.UNKNOWN;
+            WifiConfig wifiConfig = null;
+            WifiInterfaceState wifiInterfaceState = null;
+            if (type == NetInterfaceType.WIFI) {
```

Comment:
@ibinshtok I think this section of code should be simplified/split into different methods: it is hard to read/debug.